src/meta/service/src/ddl_service.rs

```diff
@@ -119,6 +122,96 @@ impl DdlServiceImpl {
             fragment_graph: fragment_graph.unwrap(),
         }
     }
+
+    pub fn start_migrate_table_fragments(&self) -> (JoinHandle<()>, Sender<()>) {
+        tracing::info!("start migrate legacy table fragments task");
+        let env = self.env.clone();
+        let metadata_manager = self.metadata_manager.clone();
+        let ddl_controller = self.ddl_controller.clone();
+
+        let (shutdown_tx, mut shutdown_rx) = tokio::sync::oneshot::channel();
+        let join_handle = tokio::spawn(async move {
+            async fn migrate_inner(
+                shutdown_rx: &mut tokio::sync::oneshot::Receiver<()>,
```

Comment:
Seems no need to pass the `shutdown_rx` here. We can do select for `shutdown_rx` with the whole other processing logic. Code will be like
```
tokio::spawn(async move {
    async fn migrate_inner() {
        while xxx {
            ... // no need to be aware of shutdown_rx here
        }
    }
    select! {
        shutdown_rx => {
               
        }
        migrate_inner() => {
            ...
        }
    }
});
```

If we don't care about the output of select, we can even do `select(shutdown_rx, migrate_inner())` directly without using the select macro.